packages/components/src/components/input/input.jsx

```diff
@@ -36,59 +41,92 @@ const Input = (
         e.target.value = input_value;
         props.onChange(e);
     };
+
+    const has_footer = has_character_counter || (hint && is_relative_hint);
+
     return (
-        <div
-            className={classNames('dc-input', className, {
-                'dc-input__disabled': disabled,
-                'dc-input--error': error,
-            })}
-        >
-            {leading_icon &&
-                React.cloneElement(leading_icon, {
-                    className: classNames('dc-input__leading-icon', leading_icon.props.className),
+        <InputWrapper has_footer={has_footer}>
+            <div
+                className={classNames('dc-input', className, {
+                    'dc-input__disabled': disabled,
+                    'dc-input--error': error,
                 })}
-            {props.type === 'textarea' ? (
-                <textarea
-                    ref={ref}
-                    {...props}
-                    className={classNames('dc-input__field dc-input__textarea', {
-                        'dc-input__field--placeholder-visible': !label && props.placeholder,
+            >
+                {leading_icon &&
+                    React.cloneElement(leading_icon, {
+                        className: classNames('dc-input__leading-icon', leading_icon.props.className),
                     })}
-                    onChange={changeHandler}
-                    disabled={disabled}
-                />
-            ) : (
-                <input
-                    ref={ref}
-                    {...props}
-                    className={classNames('dc-input__field', {
-                        'dc-input__field--placeholder-visible': !label && props.placeholder,
+                {props.type === 'textarea' ? (
+                    <textarea
+                        ref={ref}
+                        {...props}
+                        className={classNames('dc-input__field dc-input__textarea', {
+                            'dc-input__field--placeholder-visible': !label && props.placeholder,
+                        })}
+                        onChange={changeHandler}
+                        disabled={disabled}
```

Comment:
Can we use `is_disabled`?

Reply:
Let's stick to disabled considering this component is used in about 59 files.